tools/firmware-utils/src/mktplinkfw2.c

```diff
@@ -156,6 +156,12 @@ static struct flash_layout layouts[] = {
 		.kernel_ep	= 0x80000000,
 		.rootfs_ofs	= 0x140000,
 	}, {
+		.id		= "8MLmtk",
+		.fw_max_len	= 0x7b0000,
```

Comment:
Should .fw_max_len be 0x7A0000 ?
Detail: https://forum.openwrt.org/viewtopic.php?pid=339116#p339116

tl;dr: if the final sysupgrade size goes beyond 0x7A0000 bytes, it's likely can not be used to initially switch to lede/openwrt via bootloaders tftp recovery (which is the only known method). However, it still could be used to upgrade the existing openwrt/lede installation.